src/bson.rs

```diff
@@ -41,15 +41,15 @@ use crate::{
 #[derive(Clone, Debug, PartialEq)]
 pub enum Bson {
     /// 64-bit binary floating point
-    FloatingPoint(f64),
+    Double(f64),
     /// UTF-8 string
     String(String),
     /// Array
     Array(Array),
     /// Embedded document
     Document(Document),
     /// Boolean value
-    Boolean(bool),
+    Bool(bool),
```

Comment:
Why not `Boolean`, it seems pretty consistently named that in the spec as well as other drivers.

Reply:
I gave a little bit of the rationale behind how the names were chosen in the PR's description. It basically boils down to this: 
- what other drivers / the spec refers to defines the type. 
- If there is an native Rust equivalent type that corresponds to it with the same or abbreviated name, use that. 
- Otherwise, use the full name. 

So for `Double`, since there is no Rust `doub` or `double`, we just use the full `Double`. For `Boolean`, we use the Rust `Bool` terminology.

[`serde_json` similarly refers to this case as `Bool`](https://docs.serde.rs/serde_json/enum.Value.html#variant.Bool), despite the generally accepted name for the type to be "boolean" in JSON.

Reply:
According to the description your order of priority is consistency:
- with each other
- with other MongoDB drivers
- with the Rust ecosystem as a whole

I would argue that you probably want to add a "with the bson spec" somewhere next to "with other MongoDB drivers". 

Am I interpreting this list of priority backwards? Otherwise it seems like there's pretty overwhelming support for calling it `Boolean`, considering there is no mention of "bool" in the bsonspec, a quick glance at the top 4 most used drivers use the term `Boolean`. Additionally it is not a Rust `Bool`, it is a BSON wrapper of that type called `Boolean` - there is additional type information here that distinguishes it.

Reply:
The order of priorities you state is correct, but it's more of an "assembly line" type structure than discrete priorities. The "with the Rust ecosystem as a whole" part only affects the name with respect to spelling and abbreviations, not the actual name itself. It's why we name the case `Double` and not `f64` or `FloatingPoint`. It's also why I opted for `I32` over `Int32`--they're both abbreviations, so we might as well use the Rust abbreviation, right? (I do agree that it's unfortunate that the Rust language opted for nonstandard abbreviations).

So for boolean, both the spec and the other drivers refer to it as `Boolean`, which means that the type we're modeling here is named `Boolean`. Only then do we look to the Rust ecosystem to see if there's a generally accepted way to refer to boolean types (which there is, `bool`).

>Additionally it is not a Rust Bool, it is a BSON wrapper of that type called Boolean - there is additional type information here that distinguishes it.

I think this is a good point, and is sort of where I came up with my rationale as a compromise between consistency with other drivers and consistency with what Rust developers expect. (the spec/drivers defines the name, the ecosystem defines the spelling)

If that rationale all makes sense and you still think we should opt for the full types (`Boolean` / `Int32` / `Int64` / `RegularExpression`), I'm happy to make that change. I just wanted to make sure the reasoning behind the names was fully communicated before we moved off of them.

Reply:
After offline discussion, we opted for the longer names, both for clarity and to emphasize the distinction between them being BSON values rather than Rust values.